File: create_env.py
import os
import re
import socket
import subprocess
import sys

try:
    from dotenv import load_dotenv
except ImportError:
    subprocess.check_call([sys.executable, "-m", "pip", "install", "python-dotenv"])
    from dotenv import load_dotenv

def main():
    load_dotenv()
    env_file_path = os.path.join(os.getcwd(), ".env")
    env_example_file_path = os.path.join(os.getcwd(), ".env.example")
    
    with open(env_example_file_path, "r") as f:
        env_example = f.readlines()
    
    env = {}
    for key in env_example:
        key = key.strip()
        if '=' in key:
            key_name, default_value = key.split('=', 1)
            current_value = os.getenv(key_name)
            if current_value is None and "<" in default_value and ">" in default_value:
                new_value = input(f"Enter value for {key_name} (default: {default_value}): ")
                if new_value == "":
                    new_value = ""  # Save an empty string if no input is provided
                
                env[key_name] = new_value
            else:
                current_value = current_value if current_value is not None else default_value
                env[key_name] = current_value if current_value is not None else default_value

    with open(env_file_path, "w") as f:
        for key, value in env.items():
            f.write(f"{key}={value}\n")

    browser_chat_env_file_path = os.path.join(os.getcwd(), "apps", "browser-chat", ".env")
    with open(browser_chat_env_file_path, "w") as f:
        for key, value in env.items():
            f.write(f"{key}={value}\n")


if __name__ == "__main__":
    main()

File: create_env.js
const fs = require("fs");
const path = require("path");
const readline = require("readline");
require("dotenv").config();

async function main() {
  const envFilePath = path.join(process.cwd(), ".env");
  const envExampleFilePath = path.join(process.cwd(), ".env.example");

  const envExampleContent = fs.readFileSync(envExampleFilePath, "utf-8");
  const envExampleLines = envExampleContent.split("\n");

  const env = {};
  for (const line of envExampleLines) {
    const trimmedLine = line.trim();
    if (trimmedLine.includes("=")) {
      const [keyName, defaultValue = ""] = trimmedLine.split("=");
      let currentValue = process.env[keyName];

      if (
        currentValue === undefined &&
        defaultValue.includes("<") &&
        defaultValue.includes(">")
      ) {
        const new_value = await prompt(
          `Enter value for ${keyName} (default: ${defaultValue}): `
        );
        currentValue = new_value === "" ? "" : new_value;
      } else {
        currentValue = currentValue !== undefined ? currentValue : defaultValue;
      }
      env[keyName] = currentValue;
    }
  }

  saveEnvFile(env, envFilePath);

  const browserChatEnvFilePath = path.join(
    process.cwd(),
    "apps",
    "interactor",
    ".env"
  );
  saveEnvFile(env, browserChatEnvFilePath);
}

function saveEnvFile(env, filePath) {
  const content = Object.entries(env)
    .map(([key, value]) => `${key}=${value}`)
    .join("\n");
  fs.writeFileSync(filePath, content);
}

function prompt(query) {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  return new Promise((resolve) =>
    rl.question(query, (ans) => {
      rl.close();
      resolve(ans);
    })
  );
}

main()
  .then(() => console.log("Environment setup complete."))
  .catch(console.error);
